spec/datadog/tracing/contrib/rails/support/rails7.rb

```diff
@@ -0,0 +1,213 @@
+    end
+    Class.new(klass)
+  end
+
+  let(:before_test_initialize_block) do
+    proc do
+      append_routes!
+    end
+  end
+
+  let(:after_test_initialize_block) do
+    proc do
+      # Rails autoloader recommends controllers to be loaded
+      # after initialization. This will be enforced when `zeitwerk`
+      # becomes the only supported autoloader.
+      append_controllers!
+
+      # Force connection to initialize, and dump some spans
+      application_record.connection unless (defined? no_db) && no_db
+
+      # Skip default Rails exception page rendering.
+      # This avoid polluting the trace under test
+      # with render and partial_render templates for the
+      # error page.
+      #
+      # We could completely disable the {DebugExceptions} middleware,
+      # but that affects Rails' internal error propagation logic.
+      # render_for_browser_request(request, wrapper)
+      allow_any_instance_of(::ActionDispatch::DebugExceptions).to receive(:render_exception) do |this, env, exception|
+        wrapper = ::ActionDispatch::ExceptionWrapper.new(env, exception)
+
+        this.send(:render, wrapper.status_code, 'Test error response body', 'text/plain')
+      end
+    end
+  end
+
+  before do
+    reset_rails_configuration!
+  end
+
+  after do
+    reset_rails_configuration!
+
+    # Push this to base when Rails 3 removed
+    # Reset references stored in the Rails class
+    Rails.app_class = nil
+    Rails.cache = nil
+  end
+
+  def append_routes!
+    # Make sure to load controllers first
+    # otherwise routes won't draw properly.
+    test_routes = routes
+
+    rails_test_application.instance.routes.append do
+      test_routes.each do |k, v|
+        # no-dd-sa
+        if k.is_a?(Array)
+          send(k.first, k.last => v)
+        else
+          get k => v
+        end
+      end
+    end
+
+    # ActionText requires ApplicationController to be loaded since Rails 6
+    example = self
+    ActiveSupport.on_load(:action_text_content) do
+      example.stub_const('ApplicationController', Class.new(ActionController::Base))
+    end
+  end
+
+  def append_controllers!
+    controllers
+  end
+
+  # Rails leaves a bunch of global class configuration on Rails::Railtie::Configuration in class variables
+  # We need to reset these so they don't carry over between example runs
+  def reset_rails_configuration!
+    # Reset autoloaded constants
+    ActiveSupport::Dependencies.clear if Rails.application
+
+    # TODO: Remove this side-effect on missing log entries
+    Lograge.remove_existing_log_subscriptions if defined?(::Lograge)
+
+    reset_class_variable(ActiveRecord::Railtie::Configuration, :@@options) if Module.const_defined?(:ActiveRecord)
+
+    ActiveSupport::Dependencies.autoload_paths = []
+    ActiveSupport::Dependencies.autoload_once_paths = []
+    ActiveSupport::Dependencies._eager_load_paths = Set.new
+    ActiveSupport::Dependencies._autoloaded_tracked_classes = Set.new
```

Comment:
Should we also reset this class variable in rails6.rb ?

Reply:
yes, we can added, I just wanted to avoid unrelated changes in this PR, since it's pretty big already

Reply:
and I was actually wondering why tests were not failing before without this file

Reply:
apparently this is a new property in rails 7: https://app.circleci.com/pipelines/github/DataDog/dd-trace-rb/15860/workflows/e14401f3-8207-4685-ac14-804236cecf0f/jobs/575944